docs: prepare for IQP migration (#259)
Qiskit/qiskit-fermions: README.md

# Qiskit Fermions

`qiskit-fermions` extends the Qiskit SDK with tools for working on fermionic systems. Within its
scope it provides:

- efficient data structures for the representation and manipulation of fermionic operators in
  different forms
- a framework for implementing operator conversion methods (including fermion-to-qubit encodings)
- a library of efficient implementations of common conversion methods
- a framework and library of gates for expressing fermionic circuits

> - a transpilation pipeline integrated with Qiskit's transpiler process to synthesize fermionic

Reviewer: ```suggestion
- a transpilation pipeline integrated with the Qiskit transpiler process to synthesize fermionic
```
or, the `transpiler process in Qiskit` - it's one of those weird style rules that we do not add possessives to product/service names

  circuits into qubit-based circuits

Additionally, `qiskit-fermions` integrates with other tools of the ecosystem, such as:

- [ffsim](https://qiskit-community.github.io/ffsim/) for efficient simulation of fermionic circuits

----------------------------------------------------------------------------------------------------

### Documentation

[Documentation](https://quantum.cloud.ibm.com/docs/addons/qiskit-fermions) for this package is

> available on the IBM Quantum Platform.

Reviewer: ```suggestion
available on IBM Quantum Platform.
```
Also, could just remove this section altogether, since it appears within its own documentation?

Reviewer: Ignore the section removal advice!! Wrong file 😆

----------------------------------------------------------------------------------------------------

> ### Installation

Author: I think we removed it from `docs/index.rst` but kept it in the `README`

#### C

Please refer to the [C installation instructions](docs/install-c.rst).

#### Python

We encourage installing this package via `pip`, when possible:

```bash
pip install 'qiskit-fermions'
```

> For more installation information refer to these [installation instructions](docs/install.rst).

Reviewer: ```suggestion
For more installation information, refer to these [installation instructions](docs/install.rst).
```

----------------------------------------------------------------------------------------------------

### Getting started

Several guides exist to help you get started with this package. A good starting point to get an
 overview of its breadth of features is the [1D Fermi-Hubbard guide](docs/guides/flowsets.rst).

----------------------------------------------------------------------------------------------------

### Use case examples

Components of this package have been used in research related to the following
papers:

- The qDRIFT randomized circuit compilation in [^1].
- The fermion-to-qubit synthesis during the transpilation process in [^2].

----------------------------------------------------------------------------------------------------

### Technical discussion

> #### Design Intentions

Reviewer: ```suggestion
#### Design intentions
```

This package is deliberately designed to align with Qiskit: it builds on a core
implemented in Rust and provides first-party language bindings to Python and C.
Its API intends to draw parallels to Qiskit in order to seamlessly integrate
into the workflows of users with experience in programming Qiskit.

A core principal to the design of `qiskit-fermions` was the decoupling of its
fermionic circuit representation from its qubitized form. To be more precise:
the fermionic circuits are meaningful by themselves and do not require a mapping
to qubit space to be interpretable.

> Furthermore, the fermionic circuit representation may not make any assumptions

Reviewer: ```suggestion
A core principle to the design of `qiskit-fermions` was the decoupling of its
fermionic circuit representation from its qubitized form. To be more precise:
the fermionic circuits are meaningful by themselves and do not require a mapping
to qubit space to be interpretable.
Furthermore, the fermionic circuit representation cannot make any assumptions
```

about its fermion-to-qubit encoding applied later on. Consequently, while
Jordan-Wigner retains a dominant position and role, it is not assumed to be the
_default_ fermion-to-qubit encoding.

#### Known issues

> As long as Qiskit's own C API has not yet reached feature parity with its Python

Reviewer: ```suggestion
As long as the Qiskit C API has not yet reached feature parity with its Python
```

The developer guide is located at
[CONTRIBUTING.md](https://github.com/Qiskit/qiskit-fermions/blob/main/CONTRIBUTING.md)
in the root of this project's repository.
By participating, you are expected to uphold Qiskit's
[code of conduct](https://github.com/Qiskit/qiskit/blob/main/CODE_OF_CONDUCT.md).

----------------------------------------------------------------------------------------------------

### Citing this package

If you use this package in your research, use the [CITATION.bib](CITATION.bib) file in this project’s
repository to cite the appropriate reference(s).

----------------------------------------------------------------------------------------------------

### License

[Apache License 2.0](LICENSE.txt)

----------------------------------------------------------------------------------------------------

> ### Deprecation Policy

Reviewer: ```suggestion
### Deprecation policy
```
